### PR TITLE
Implement simple cross mm config apis

### PR DESCRIPTION
**Pull request by haoze-grvt**
- Separate simple cross maintenance margin config from other config
    - Add separate state for simple cross mm configs
    - Add separate schedule and set methods for simple cross mm configs(so that they can be updated atomically)
- Change simple cross mm configs from per-underlying-currency to per-kind-underlying-quote(KUQ)
- Change simple cross mm calculation logic from simple lookup to progressive tax like algorithm

Fixes https://cantina.xyz/code/a72be44c-8600-4faa-9081-277c561ca36a/findings/33

## Progress Margin Tier calculation example
```
0+ size, 1%
10+ size, 1.5%
100+ size, 2%
User A has size 14
Effective Margin Rate = (1% * 10 + 1.5% * 4) / 14 = 1.143%
User B has size 125
Effective Margin Rate = (1% * 10 + 1.5% * 90 + 2% * 25) / 125 = 1.56%
```

## Simple Cross Maintenance Margin Tiers Timelock Rule
- 2 week timelock required if **a higher maintenance margin ratio is required for any size**
- no timelock otherwise

### Logic
Each `ListMarginTiersBI` defines a piecewise linear function `_calculateSimpleCrossMMSize`
- which maps position size to maintenance margin requirement
    - `function _calculateSimpleCrossMMSize(ListMarginTiersBI memory mt, BI memory size) internal pure returns (BI memory)`
    - where `_calculateSimpleCrossMMSize(mt, 0) = 0`
    - where the gradient only increase after each breakpoint(bracketStart)

This function finds whether the other MarginTiersBI has a higher margin requirement at any position size. The idea is to
 - Iterate through all bracketStart in the 2 MarginTiersBIs for a point where the other tiers has higher requirement
 - Check the rate of the last tier
     - the tiers with higher last tier rate will have higher requirement with large enough size

### CODE REVIEW

File: contracts/exchange/api/RiskCheck.sol [hunk 1]
```diff
@@ -179,24 +104,19 @@ contract RiskCheck is BaseContract, ConfigContract {
...
       int64 size = values[id].balance;
       if (size < 0) {
         size = -size;
       }
-      BI memory sizeBI = BI(int256(size), _getBalanceDecimal(underlyingCurrency));
-      uint mmConfigIdx = uint(underlyingCurrency) - uint(Currency.ETH);
-      BI memory ratio = _getMaintenanceMarginRatio(sizeBI, mmConfigs[mmConfigIdx]);
-      // The charge for a perpetual positions = (Size) * (Mark Price) * (Maintenance Ratio)
-      BI memory charge = ratio.mul(_requireMarkPriceBI(id)).mul(sizeBI);
+      bytes32 kud = assetGetKUQ(id);
```
Comment (minhbsq) [inline contracts/exchange/api/RiskCheck.sol:116]: typo: kuq

File: contracts/exchange/types/DataStructure.sol [hunk 2]
```diff
@@ -55,6 +56,7 @@ function currencyIsValid(Currency iter) pure returns (bool) {
 
 uint constant PRICE_DECIMALS = 9;
 uint constant CENTIBEEP_DECIMALS = 6;
+uint constant BPS_DECIMALS = 4;
```
Comment (minhbsq) [inline contracts/exchange/types/DataStructure.sol:59]: use a non-abbreviated name for clarity

File: contracts/exchange/types/DataStructure.sol [hunk 3]
```diff
@@ -309,6 +302,21 @@ struct ConfigSetting {
...
+  uint64 bracketStart;
+  uint32 rate;
+}
+
+struct MarginTierBI {
+  BI bracketStart;
+  BI rate;
+}
+
+struct MarginTiersBI {
```
Comment (minhbsq) [inline contracts/exchange/types/DataStructure.sol:315]: nit: `ListMarginTiersBI`

File: contracts/exchange/types/DataStructure.sol (hunk 2 above)
Comment (haoze-grvt) [reply contracts/exchange/types/DataStructure.sol]: done

File: contracts/exchange/api/RiskCheck.sol (hunk 1 above)
Comment (haoze-grvt) [reply contracts/exchange/api/RiskCheck.sol:116]: fixed

File: contracts/exchange/types/DataStructure.sol (hunk 3 above)
Comment (haoze-grvt) [reply contracts/exchange/types/DataStructure.sol]: Renamed

**Review by minhbsq — APPROVED** [review]
lgtm

**Pull request merged**